### ISSUE TITLE
Lack of type hints

### ISSUE THREAD

**Issue by vemv** [clj-time/clj-time#268]
Hi,

first, just to give some context, generally at work we have the policy of delivering clj code with zero reflection warnings.

Now, `clj-time` makes this goal somewhat repetitive because it lacks some annotations which would seem trivial/low-cost to add.

e.g., this change could be introduced:

```diff
(defn now
  "Returns a DateTime for the current instant in the UTC time zone."
-  []
+ ^DateTime []
  (DateTime. ^DateTimeZone utc))
```

And so on.

Don't know if this has been considered already. WDYT?

**Comment by seancorfield** [comment]
Several of the functions _do_ have type hints on their return values so I wouldn't object to a PR that adds them in other places. 

However...

At this point, `clj-time` is in maintenance-only mode per the README:

> If you are using Java 8 or later, consider using the built-in Java Time instead of Joda Time -- and look at clojure.java-time if you want a Clojure wrapper for that, or cljc.java-time for a thin Clojure(Script) wrapper. See Converting from Joda Time to java.time for more details about the similarities and differences between the two libraries.

If you still want type hints added, given that caveat, feel free to submit a PR.

**Comment by vemv** [comment]
Thank you!

I appreciate the confirmation and heads-up.

Will proceed.

**Comment by seancorfield** [comment]
Addressed in #269
